src/Skipper/Skipper/Skipper.php

```diff
@@ -49,9 +50,10 @@ public function shouldSkipCurrentNode(
         string $filePath,
         string $rectorClass,
         Node $node
-    ): bool {
-        if ($this->shouldSkipElementAndFilePath($element, $filePath)) {
-            return true;
+    ): bool|int {
+        if ($this->shouldSkipElementAndFilePath($element, $filePath, $node)) {
+            // Don't go any deeper with this rule if we already know we will skip:
+            return NodeVisitor::DONT_TRAVERSE_CHILDREN;
```

Comment:
use bool return only on non-visitor, return dont traverse children is only allowed on callable traverser, and refactor() method, which handled by AbstractRector, to allow other rules execute

https://github.com/rectorphp/rector-src/blob/1ec42bcc33f0607533e08d75c47b589d89e7649e/src/Rector/AbstractRector.php#L292-L314

Reply:
What do you mean with "only allowed"? This is internal code :/

Using a boolean it is not possible to distinguish whether to skip the children or not.

Reply:
If can't be bool, then null should be the way. Skipper is not visitor.

Reply:
Ok, I can use null.

Reply:
Hmm, I felt uncomfortable with using `null` as a truthy skip result. I'm using an enum now: 

```
enum SkipperResult
{
    case noSkip;
    case skipNode;
    case skipTree;
}
```

which is converted to a NodeVisitor::XXX result in the AbstractRector.

Reply:
Don't directly return `NodeVisitor::DONT_TRAVERSE_CHILDREN` on `AbstractRector` as it will skip next all rules after it to visit the node, it only allowed on child of Rector rule `refactor()` which verifiable by AbstractRector to set attribute skip only on rule after it

https://github.com/rectorphp/rector-src/blob/912d0b7ecc78aae4cbc2f3df7992fab9b5c63014/src/Rector/AbstractRector.php#L156-L173